sdcm/cluster_aws.py

```diff
@@ -120,14 +129,42 @@ def add_nodes(self, count, ec2_user_data='', dc_idx=0):
-                                                                UserData=ec2_user_data,
-                                                                MinCount=count,
-                                                                MaxCount=count,
-                                                                KeyName=self._credentials[dc_idx].key_pair_name,
-                                                                BlockDeviceMappings=self._ec2_block_device_mappings,
-                                                                NetworkInterfaces=interfaces,
-                                                                InstanceType=self._ec2_instance_type)
+
+        if self.instance_provision == INSTANCE_PROVISION_ON_DEMAND:
+            instances = self._ec2_services[dc_idx].create_instances(ImageId=self._ec2_ami_id[dc_idx],
+                                                                    UserData=ec2_user_data,
+                                                                    MinCount=count,
+                                                                    MaxCount=count,
+                                                                    KeyName=self._credentials[dc_idx].key_pair_name,
+                                                                    BlockDeviceMappings=self._ec2_block_device_mappings,
+                                                                    NetworkInterfaces=interfaces,
+                                                                    InstanceType=self._ec2_instance_type)
+        else:
+            ec2 = ec2_client.EC2Client()
+            subnet_info = ec2.get_subnet_info(self._ec2_subnet_id[dc_idx])
+            spot_params = dict(instance_type=self._ec2_instance_type,
+                               image_id=self._ec2_ami_id[dc_idx],
+                               region_name=subnet_info['AvailabilityZone'],
+                               network_if=interfaces,
+                               key_pair=self._credentials[dc_idx].key_pair_name,
+                               user_data=ec2_user_data,
+                               count=count)
+
+            if self.instance_provision == INSTANCE_PROVISION_SPOT_FLEET and count > 1:
+                request_cnt = 1
+                if count > SPOT_FLEET_LIMIT:
+                    request_cnt = count / SPOT_FLEET_LIMIT
+                    spot_params['count'] = SPOT_FLEET_LIMIT
+                instances = []
+                for i in range(request_cnt):
+                    instances_i = ec2.create_spot_fleet(**spot_params)
+                    instances.extend(instances_i)
+            elif self.instance_provision == INSTANCE_PROVISION_SPOT_LOW_PRICE or count == 1:
+                instances = ec2.create_spot_instances(**spot_params)
+            elif self.instance_provision == INSTANCE_PROVISION_SPOT_DURATION:
+                spot_params.update({'duration': cluster.TEST_DURATION / 60 * 60})
```

Comment:
In case the test duration is not full hour for example 2.5 hours, the duration will be 2 hours which is not enough.
It's better to have more than less...

Reply:
I'll add one more hour